### DISCUSSION TITLE
Are bitmap fonts supported on Windows?

### DISCUSSION THREAD

**Discussion by carlbolduc** [wezterm/wezterm#3071] [Q&A] [answered]
I installed the Terminus font on my Windows 11 PC. I can see that Terminus is available when I go into the fonts section of the control panel (it's a .fon file, not a .ttf file). However, I cannot use in in Wezterm. It looks like Wezterm is only seeing the TTF fonts. Is it possible to use bitmap fonts on Windows?

**Comment by wez** [comment, marked as answer]
wezterm supports bitmap fonts. I've personally used the Terminus OTB (Open Type Bitmap) font with wezterm.

wezterm uses freetype to load fonts, and that appears have support for loading the `.fnt` data out of a `.fon` container.

On Windows, wezterm can query the GDI API and the more modern DirectWrite API to discover installed fonts and access their data to pass into freetype but both of those APIs work with True Type file formats and don't support the very old `FNT`/`FON` data.

I think you might have two options to get this working on your windows system:

## font_dirs

Try setting [font_dirs](https://wezfurlong.org/wezterm/config/lua/config/font_dirs.html) up and listing the directory where the `.fon` file(s) are found.  That will cause wezterm to scan the listed locations for font files and attempt to load them with freetype.

You'll be able to tell if this worked if `wezterm ls-fonts --list-system` shows you the `.fon` files

## Obtain or Build the OTB files

On a unix system (eg: WSL) you can download the sources for Terminus from https://terminus-font.sourceforge.net/ and extract it and build the OTB files; it only takes a few seconds to run:

```
./configure
make otb
```

you can then copy and install the `.otb` files

On a debian system, you can install `fonts-terminus-otb` and copy the otb files from there.  Other distributions may have a different package name for the same files.

**Reply by carlbolduc** [reply]
Thank you for the detailed answer, adding `font_dirs = { 'c:/Windows/Fonts' },` to my config allowed me to use Terminus.

Wezterm is a superb terminal, I'm glad I made the effort to create a config file and took some time to read the doc.

Thank you for your work.
